collect_app/src/main/java/org/odk/collect/android/widgets/AbstractDateWidget.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2017 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.widgets;
+
+import android.content.Context;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import org.javarosa.core.model.data.DateData;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.joda.time.LocalDateTime;
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.logic.DatePickerDetails;
+import org.odk.collect.android.logic.FormController;
+import org.odk.collect.android.utilities.DateTimeUtils;
+
+/**
+ * @author Grzegorz Orczykowski (gorczykowski@soldevelo.com)
+ */
+public abstract class AbstractDateWidget extends QuestionWidget implements BinaryWidget {
+
+    protected Button dateButton;
+    protected TextView dateTextView;
+
+    protected boolean nullAnswer;
```

Comment:
Can you add a comment describing what this is?

Reply:
Or maybe just rename it `isAnswerNull`?

Reply:
Fixed.